processor/src/main/java/org/derive4j/processor/derivator/LazyConstructorDerivator.java

```diff
@@ -73,20 +73,20 @@
                 .addModifiers(Modifier.PRIVATE)
                 .returns(typeName)
                 .addCode(CodeBlock.builder()
-                    .addStatement("$T _evaluation = this.evaluation", typeName)
-                    .beginControlFlow("if (_evaluation == null)")
-                    .beginControlFlow("synchronized (this.lock)")
```

Comment:
I did not synchronizing on ```this``` to avoid potential synchronization issues as the lazy value could potentially be used as a monitor. But then this would probably be bad practice. As a reference, I think Guava Suppliers.memoize does synchronize on ```this```. I need to benchmark if avoiding the creation of the lock Object result in significative performance improvement.